bitarray.c

```diff
@@ -0,0 +1,25 @@
+#include "bitarray.h"
+
+Bitarray init_zeroed_bitarray(void)
+{
+	Bitarray zeroed = {0};
+	return zeroed;
+}
+
+void write_bit(Bitarray* bitarray, size_t indx, bool bit)
+{
+	size_t long_offset = indx / BITS_PER_LONG;
+	unsigned char bit_offset = indx % BITS_PER_LONG;
+	long mask = bit << (BITS_PER_LONG - bit_offset - 1);
+
+	bitarray->data[long_offset] |= mask;
+}
+
+bool read_bit(const Bitarray* bitarray, size_t indx)
+{
+	size_t long_offset = indx / BITS_PER_LONG;
+	unsigned char bit_offset = indx % BITS_PER_LONG;
+	long mask = 1 << (BITS_PER_LONG - bit_offset - 1);
+
+	return (bitarray->data[long_offset] & mask) != 0;
+}
```

Comment:
**suggestion (blocking):** Is there a description we can add to explain the operations that are happening in these methods? A link to an explanation would suffice also.

Reply:
I added some explanatory comments, although this is a pretty standard way of reading/writing bits so I wouldn't normally say it should be documented